L1Trigger/CSCTriggerPrimitives/src/CSCAnodeLCTProcessor.cc

```diff
@@ -295,6 +295,11 @@ void CSCAnodeLCTProcessor::run(const std::vector<int> wire[CSCConstants::NUM_LAY
   // Take the best MAX_CLCTS_PER_PROCESSOR candidates per bx.
   int ALCTIndex_[CSCConstants::MAX_ALCT_TBINS] = {};
 
+  // define a new pattern map
+  // for each key half strip, and for each pattern, store the 2D collection of fired comparator digis
+  std::map<int, std::map<int, CSCCLCTDigi::ComparatorContainer>> hits_in_patterns;
```

Comment:
so this is a map of a map of a vector of vectors?

Reply:
@davidlange6 So the type needs to be corrected (https://github.com/cms-sw/cmssw/pull/30940), but it shouldn't have any effect. ComparatorContainer and WireContainer are both 2D vectors.

Yes, it maps wiregroups and pattern numbers onto 2D vectors with hits.